src/annotator/util/emitter.ts

```diff
@@ -1,54 +1,48 @@
+/* eslint-disable @typescript-eslint/ban-types */
+
+/*
+ * Disable @typescript-eslint/ban-types for the whole file, as changing the
+ * event's callback type away from `Function` has multiple implications that
+ * should be addressed separately
+ */
```

Comment:
I have disabled `@typescript-eslint/ban-types` for the whole type because it complains everywhere where `Function` is used.

I have tried replacing it with something else, like `(...args, unknown[]) => unknown`, but then there are type errors in multiple places.

We probably need to give it another thought, but I would like to do that on a follow-up PR.